azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -272,72 +282,106 @@ public int getHttpProxyPort() {
+
         try {
-            final AzureEnvironment environment;
-            if (!AzureAuthHelper.validateEnvironment(auth.getEnvironment())) {
-                Log.prompt(String.format(UNSUPPORTED_AZURE_ENVIRONMENT, auth.getEnvironment()));
-                environment = AzureEnvironment.AZURE;
-            } else {
-                environment = AzureAuthHelper.getAzureEnvironment(auth.getEnvironment());
+            final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
+            mavenAuthConfiguration.setType(getAuthType());
+            azureCredentialWrapper = MavenAuthUtils.login(session, settingsDecrypter, mavenAuthConfiguration);
+            if (Objects.isNull(azureCredentialWrapper)) {
+                return null;
             }
-            final String environmentName = AzureAuthHelper.getAzureEnvironmentDisplayName(environment);
-            if (environment != AzureEnvironment.AZURE) {
-                Log.prompt(String.format(USING_AZURE_ENVIRONMENT, environmentName));
+
+            final AzureEnvironment env = azureCredentialWrapper.getEnv();
+            final String environmentName = AuthHelper.azureEnvironmentToString(env);
+            if (env != AzureEnvironment.AZURE) {
+                Log.prompt(String.format(USING_AZURE_ENVIRONMENT, TextUtils.green(environmentName)));
+            }
+            Log.info(azureCredentialWrapper.getCredentialDescription());
+            final Azure tempAzure = Azure.configure()
+                    .authenticate(azureCredentialWrapper.getAzureTokenCredentials()).withDefaultSubscription();
+            final PagedList<Subscription> subscriptions = tempAzure.subscriptions().list();
+            subscriptions.loadAll();
+            final List<String> subsIdList = subscriptions.stream().map(Subscription::subscriptionId).collect(Collectors.toList());
+            String defaultSubscriptionId = StringUtils.firstNonBlank(this.subscriptionId, azureCredentialWrapper.getDefaultSubscriptionId());
+
+            if (StringUtils.isBlank(defaultSubscriptionId) && ArrayUtils.isNotEmpty(azureCredentialWrapper.getFilteredSubscriptionIds())) {
+                final Collection<String> filteredSubscriptions = StringListUtils.intersectIgnoreCase(subsIdList,
+                        Arrays.asList(azureCredentialWrapper.getFilteredSubscriptionIds()));
+                if (filteredSubscriptions.size() == 1) {
+                    defaultSubscriptionId = filteredSubscriptions.iterator().next();
+                }
+            }
+            if (StringUtils.isBlank(defaultSubscriptionId) && subsIdList.size() == 1) {
+                defaultSubscriptionId = subsIdList.get(0);
+            }
+
+            if (StringUtils.isAllBlank(this.subscriptionId, defaultSubscriptionId)) {
+                defaultSubscriptionId = selectSubscription(tempAzure, subscriptions.toArray(new Subscription[0]));
```

Comment:
we have some duplicated codes for default subscription like choose the only subs id in both here and `selectSubscription`, throw exception when no subs in `checkSubscription` and `selectSubscription`, can we remove the duplicated check?